wen-restart/src/last_voted_fork_slots_aggregate.rs

```diff
@@ -90,80 +125,131 @@ impl LastVotedForkSlotsAggregate {
         &mut self,
         new_slots: RestartLastVotedForkSlots,
     ) -> Option<LastVotedForkSlotsRecord> {
-        let total_stake = self.epoch_stakes.total_stake();
-        let threshold_stake = (total_stake as f64 * self.repair_threshold) as u64;
         let from = &new_slots.from;
         if from == &self.my_pubkey {
             return None;
         }
-        let sender_stake = Self::validator_stake(&self.epoch_stakes, from);
-        if sender_stake == 0 {
-            warn!(
-                "Gossip should not accept zero-stake RestartLastVotedFork from {:?}",
-                from
-            );
-            return None;
-        }
         self.active_peers.insert(*from);
-        let new_slots_vec = new_slots.to_slots(self.root_slot);
+        let root_slot = self.root_bank.slot();
+        let new_slots_vec = new_slots.to_slots(root_slot);
         let record = LastVotedForkSlotsRecord {
             last_voted_fork_slots: new_slots_vec.clone(),
             last_vote_bankhash: new_slots.last_voted_hash.to_string(),
             shred_version: new_slots.shred_version as u32,
             wallclock: new_slots.wallclock,
         };
+        if !self.update_slots_stakes_map_and_slots_to_repair(new_slots, new_slots_vec) {
+            return None;
+        }
+        self.update_epoch_info_vec();
+        Some(record)
+    }
+
+    fn update_slots_stakes_map_and_slots_to_repair(
+        &mut self,
+        new_slots: RestartLastVotedForkSlots,
+        new_slots_vec: Vec<Slot>,
+    ) -> bool {
```

Comment:
If the purpose of this boolean is to know when to run `update_epoch_info_vec`, seems to make more sense to return an `Option<Slot>` that indicates if we hit the repair threshold for a new slot.

We can return  `Some(new_latest_repair_slot)` when we insert into `self.slots_to_repair.insert(*slot);`, or `None`.

Then `update_epoch_info_vec()` can just take as an argument the new latest slot and we don't have to do a `match highest_repair_slot` at all in `update_epoch_info_vec()`

Reply:
Sorry that's bad naming, it just returns whether the given new slots has already been saved before, if it has been saved before then we can short circuit and avoid further updates.

Renamed, does this look better?